src/storage/rocksdb.cpp

```diff
@@ -255,6 +255,39 @@ void RocksDBStorageProvider::endWriteBatch()
     m_lock.unlock();
 }
 
+struct BatchStorageToken : public StorageToken {
+    std::shared_ptr<rocksdb::DB> tspdb;    // Note: This must be first so it is deleted last
+    rocksdb::WriteBatch* tspbatch;
```

Comment:
naming convention , make it unique_ptr and transfer ownership and release lock in complete